crates/flux-middle/src/rty/expr.rs

```diff
@@ -647,7 +653,13 @@ impl Ctor {
     }
 }
 
-#[derive(Clone, PartialEq, Eq, Hash, TyEncodable, TyDecodable)]
+#[derive(Clone, PartialEq, Eq, Hash, TyEncodable, Debug, TyDecodable)]
+pub enum QuantKind {
+    Exists,
+    Forall,
+}
```

Comment:
Let's reuse the `fhir` one here. It's okay for `rty` to depend on `fhir` but not the other way around. You'll have to derive the necessary things on the `fhir` one. Also, derive `Encodable/Decodable` instead of `TyEncodable/TyDecodable`

```suggestion
```